Automattic/woocommerce-payments
# Fix untranslated function esc_html_e

All instances of [esc_html_e](https://developer.wordpress.org/reference/functions/esc_html_e/) in the codebase are not being translated. See https://github.com/Automattic/woocommerce-payments/search?q=esc_html_e for a list of all uses.

The Babel Plugin Makepot package has [a default set of functions](https://github.com/WordPress/gutenberg/blob/85a3d565d27b1788f73fac149af49fa523986b64/packages/babel-plugin-makepot/index.js#L66-L71) to parse when making `woocommerce-payments.pot` and `esc_html_e` is not included.

```js
const DEFAULT_FUNCTIONS = {
	__: [ 'msgid' ],
	_n: [ 'msgid', 'msgid_plural' ],
	_x: [ 'msgid', 'msgctxt' ],
	_nx: [ 'msgid', 'msgid_plural', null, 'msgctxt' ],
};
```

#### Changes proposed in this Pull Request

Fix untranslated function `esc_html_e` by adding it to a `functions` property passed to `@wordpress/babel-plugin-makepot` in Babel configs. 

Another alternative would be to change uses of `esc_html_e( 'string' )` to `esc_htm( __( 'string' ) )` so that the translation is parsed correctly, but that won't prevent future breakage.

#### Testing instructions

* Confirm the problem exists on the `develop` branch by using any instance of `esc_html_e`, like this one for example https://github.com/Automattic/woocommerce-payments/blob/86def0fce907c8718fde500208718419beab6e72/includes/class-wc-payments-payment-request-button-handler.php#L700
* On `develop` run `npm run build:client` and navigate to `languages/woocommerce-payments.pot`. Search for `includes/class-wc-payments-payment-request-button-handler.php:700` and confirm the translation isn't parsed.
* Now on this branch repeat `npm run build:client` and perform the same search. You should see the following:

```
#: includes/class-wc-payments-payment-request-button-handler.php:700
msgid "OR"
msgstr ""
```

-------------------


**Post merge**
